drivers/video/video_arducam_mega.c

```diff
@@ -0,0 +1,1389 @@
+		break;
+	default:
+		return -ENODEV;
+	}
+	drv_data->camera_id = cam_id;
+
+	return ret;
+}
+
+static int arducam_mega_set_fmt(const struct device *dev, struct video_format *fmt)
+{
+	struct arducam_mega_data *drv_data = dev->data;
+	int ret = 0;
+	int i = 0;
+
+	if (!memcmp(&drv_data->fmt, fmt, sizeof(drv_data->fmt))) {
+		/* nothing to do */
+		return 0;
+	}
+
+	ret = video_format_caps_index(fmts, fmt, &i);
+	if (ret < 0) {
+		LOG_ERR("Unsupported pixel format or resolution %s %ux%u",
+			VIDEO_FOURCC_TO_STR(fmt->pixelformat), fmt->width, fmt->height);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int arducam_mega_get_fmt(const struct device *dev, struct video_format *fmt)
+{
+	struct arducam_mega_data *drv_data = dev->data;
+
+	*fmt = drv_data->fmt;
+
+	return 0;
+}
+
+static void arducam_mega_stream_schedule(struct k_timer *timer)
+{
+	struct arducam_mega_data *drv_data = timer->user_data;
+
+	k_work_submit_to_queue(&ac_work_q, &drv_data->buf_work);
+}
+
+static int arducam_mega_stream_start(const struct device *dev, bool enable,
+				     enum video_buf_type type)
+{
+	struct arducam_mega_data *drv_data = dev->data;
+
+	if (drv_data->stream_on) {
+		return 0;
+	}
+
+	if (enable) {
+		drv_data->stream_on = 1;
+		drv_data->fifo_length = 0;
+		k_timer_start(&drv_data->stream_schedule_timer, K_MSEC(30), K_MSEC(30));
+	} else {
+		drv_data->stream_on = 0;
+		k_timer_stop(&drv_data->stream_schedule_timer);
+	}
+
+	return 0;
+}
+
+static int arducam_mega_stream_stop(const struct device *dev)
+{
+	struct arducam_mega_data *drv_data = dev->data;
+
+	drv_data->stream_on = 0;
+
+	k_timer_stop(&drv_data->stream_schedule_timer);
+
+	return 0;
+}
+
+static int arducam_mega_flush(const struct device *dev, bool cancel)
+{
+	struct arducam_mega_data *drv_data = dev->data;
+	struct video_buffer *vbuf;
+
+	/* Clear fifo cache */
+	while (!k_fifo_is_empty(&drv_data->fifo_out)) {
+		vbuf = k_fifo_get(&drv_data->fifo_out, K_USEC(10));
+		if (vbuf != NULL) {
+			k_fifo_put(&drv_data->fifo_in, vbuf);
+		}
+	}
+	return 0;
+}
+
+static int arducam_mega_soft_reset(const struct device *dev)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+	struct arducam_mega_data *drv_data = dev->data;
+	int ret = 0;
+
+	if (drv_data->stream_on) {
+		arducam_mega_stream_stop(dev);
+	}
+	/* Initiate system reset */
+	ret = arducam_mega_write_reg(&cfg->bus, CAM_REG_SENSOR_RESET, SENSOR_RESET_ENABLE);
+	if (ret < 0) {
+		LOG_ERR("Failed to reset the sensor (%d)", ret);
+		return ret;
+	}
+
+	k_msleep(1000);
+
+	return ret;
+}
+
+static int arducam_mega_capture(const struct device *dev, uint32_t *length)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+	struct arducam_mega_data *drv_data = dev->data;
+	int tries = 200;
+	int ret = 0;
+	uint32_t reg_data;
+
+	arducam_mega_write_reg(&cfg->bus, ARDUCHIP_FIFO, FIFO_CLEAR_ID_MASK);
+	arducam_mega_write_reg(&cfg->bus, ARDUCHIP_FIFO, FIFO_START_MASK);
+
+	for (; tries > 0; tries--) {
```

Comment:
You could keep it a bit simpler this way:

```suggestion
	for (int tries = 200; tries > 0; tries--) {
```

And it's common practice to keep all arbitrary numbers behind `#defines`, so:

```c
#define ARDUCAM_MEGA_CAPTURE_TRIES 200
```
Then:
```suggestion
	for (int tries = ARDUCAM_MEGA_CAPTURE_TRIES; tries > 0; tries--) {
```